File: mobile/src/main/java/guepardoapps/whosbirthday/activities/ActivityBoot.java
package guepardoapps.whosbirthday.activities;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;

import guepardoapps.whosbirthday.R;
import guepardoapps.whosbirthday.common.*;
import guepardoapps.whosbirthday.controller.*;
import guepardoapps.whosbirthday.model.BirthdayDto;
import guepardoapps.whosbirthday.tools.Logger;

public class ActivityBoot extends Activity {
    private static final String TAG = ActivityBoot.class.getSimpleName();
    private Logger _logger;

    private Context _context;
    private DatabaseController _databaseController;
    private NavigationController _navigationController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.side_boot);

        _logger = new Logger(TAG, Enables.LOGGING);
        _logger.Debug("onCreate");

        _context = this;

        _databaseController = DatabaseController.getInstance();
        _databaseController.Initialize(_context);

        _navigationController = new NavigationController(_context);
        SharedPrefController sharedPrefController = new SharedPrefController(_context, SharedPrefConstants.SHARED_PREF_NAME);

        SharedPreferences initialStart = getSharedPreferences(SharedPrefConstants.SHARED_PREF_NAME, 0);
        boolean dataInstalled = initialStart.getBoolean(SharedPrefConstants.SHARED_PREF_NAME, false);
        if (!dataInstalled) {
            _databaseController.CreateBirthday(new BirthdayDto(0, "Jonas Schubert", 2, 1, 1990));
            sharedPrefController.SaveBooleanValue(SharedPrefConstants.SHARED_PREF_NAME, true);
        }

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                _navigationController.NavigateTo(ActivityMain.class, true);
            }
        }, 1500);
    }

    @Override
    protected void onPause() {
        super.onPause();
        _logger.Debug("onPause");
        _databaseController.Dispose();
    }

    @Override
    protected void onResume() {
        super.onResume();
        _logger.Debug("onResume");
        _databaseController.Initialize(_context);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        _logger.Debug("onDestroy");
        _databaseController.Dispose();
    }
}

File: mobile/src/main/java/guepardoapps/whosbirthday/activities/ActivityBoot.kt
package guepardoapps.whosbirthday.activities

import android.app.Activity
import android.os.Bundle
import android.os.Handler
import com.github.guepardoapps.kulid.ULID
import com.github.guepardoapps.timext.kotlin.extensions.millis
import com.github.guepardoapps.timext.kotlin.postDelayed
import guepardoapps.whosbirthday.R
import guepardoapps.whosbirthday.controller.NavigationController
import guepardoapps.whosbirthday.controller.SharedPreferenceController
import guepardoapps.whosbirthday.controller.SystemInfoController
import guepardoapps.whosbirthday.database.birthday.DbBirthday
import guepardoapps.whosbirthday.model.Birthday

@ExperimentalUnsignedTypes
class ActivityBoot : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.side_boot)
        val sharedPreferenceController = SharedPreferenceController(this)

        if (!sharedPreferenceController.load(getString(R.string.sharedPrefName), false)) {
            sharedPreferenceController.run {
                save(getString(R.string.sharedPrefBubbleState), true)
                save(getString(R.string.sharedPrefBubblePosX), resources.getInteger(R.integer.sharedPrefBubbleDefaultPosX))
                save(getString(R.string.sharedPrefBubblePosY), resources.getInteger(R.integer.sharedPrefBubbleDefaultPosY))
                save(getString(R.string.sharedPrefName), true)
            }

            DbBirthday(this)
                    .add(Birthday(
                            id = ULID.random(),
                            name = "Jonas Schubert",
                            group = getString(R.string.friends),
                            day = 2, month = 1, year = 1990,
                            remindMe = true, remindedMe = false))
        }
    }

    override fun onResume() {
        super.onResume()

        val navigationController = NavigationController(this)
        val systemInfoController = SystemInfoController(this)

        if (systemInfoController.currentAndroidApi() >= android.os.Build.VERSION_CODES.M) {
            if (systemInfoController.checkAPI23SystemPermission(resources.getInteger(R.integer.systemPermissionId))) {
                Handler().postDelayed({ navigationController.navigate(ActivityMain::class.java, true) }, resources.getInteger(R.integer.bootNavigationDelayInMs).millis)
            }
        } else {
            Handler().postDelayed({ navigationController.navigate(ActivityMain::class.java, true) }, resources.getInteger(R.integer.bootNavigationDelayInMs).millis)
        }
    }
}